Java/Unit/UnitIsDeadException.java

```diff
@@ -0,0 +1,10 @@
+/**
+ * Created by andrew on 04.12.16.
+ */
+public class UnitIsDeadException extends Exception {
+    public String text;
+
+    public UnitIsDeadException(String text) {
+        this.text = text;
+    }
```

Comment:
You should call super(text) which will create parent class. Also you don't need text field